cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -335,26 +335,10 @@ func (scaleSet *ScaleSet) GetScaleSetVms() ([]string, error) {
 // It is assumed that cloud provider will not delete the existing nodes if the size
 // when there is an option to just decrease the target.
 func (scaleSet *ScaleSet) DecreaseTargetSize(delta int) error {
-	if delta >= 0 {
-		return fmt.Errorf("size decrease size must be negative")
-	}
-
-	size, err := scaleSet.GetScaleSetSize()
-	if err != nil {
-		return err
-	}
-
-	nodes, err := scaleSet.Nodes()
-	if err != nil {
-		return err
-	}
-
-	if int(size)+delta < len(nodes) {
-		return fmt.Errorf("attempt to delete existing nodes targetSize:%d delta:%d existingNodes: %d",
-			size, delta, len(nodes))
-	}
-
-	scaleSet.SetScaleSetSize(size + int64(delta))
+	// The function is not needed because the target size will be good
+	// after the next time's refresh of cache. To prevent some unexpected
+	// errors (see https://github.com/kubernetes/autoscaler/issues/2594 and
+	// https://github.com/kubernetes/kubernetes/issues/83073) we clear it.
```

Comment:
replace the comment with:

```suggestion
	// VMSS size would be changed automatically after the Node deletion, hence this operation is not required.
```